src/components/User.vue

```diff
@@ -0,0 +1,224 @@
+						v-model="password"
+						:type="passwordInputType"
+						class="input__field"
+						name="password"
+						:placeholder="t('registration', 'Password')"
+						required>
+					<label for="password" class="infield">{{ t('registration', 'Password') }}</label>
+					<Lock :size="20" class="input__icon" />
+					<Button class="toggle-password"
+						type="tertiary-no-background"
+						:aria-label="isPasswordHidden ? t('registration', 'Show password') : t('registration', 'Hide password')"
+						@click.stop.prevent="togglePassword"
+						@keydown.enter="togglePassword">
+						<template #icon>
+							<Eye v-if="isPasswordHidden" :size="20" />
+							<EyeOff v-else :size="20" />
+						</template>
+					</Button>
+				</p>
+				<Button id="submit"
+					native-type="submit"
+					type="primary"
+					:wide="true"
+					:disabled="submitting || password.length === 0"
+					@click="submit">
+					{{ t('registration', 'Create account') }}
+				</Button>
+			</fieldset>
+		</form>
+	</div>
+</template>
+
+<script>
+import { getRequestToken } from '@nextcloud/auth'
+import Button from '@nextcloud/vue/dist/Components/Button'
+import { loadState } from '@nextcloud/initial-state'
+import Eye from 'vue-material-design-icons/Eye'
+import EyeOff from 'vue-material-design-icons/EyeOff'
+import Email from 'vue-material-design-icons/Email'
+import Lock from 'vue-material-design-icons/Lock'
+import Phone from 'vue-material-design-icons/Phone'
+import Account from 'vue-material-design-icons/Account'
+import Key from 'vue-material-design-icons/Key'
+
+export default {
+	name: 'User',
+
+	components: {
+		Button,
+		Eye,
+		EyeOff,
+		Email,
+		Lock,
+		Phone,
+		Account,
+		Key,
+	},
+
+	data() {
+		return {
+			email: loadState('registration', 'email'),
+			emailIsLogin: loadState('registration', 'emailIsLogin'),
+			emailIsOptional: loadState('registration', 'emailIsOptional'),
+			loginname: loadState('registration', 'loginname'),
+			fullname: loadState('registration', 'fullname'),
+			showFullname: loadState('registration', 'showFullname'),
+			enforceFullname: loadState('registration', 'enforceFullname'),
+			phone: loadState('registration', 'phone'),
+			showPhone: loadState('registration', 'showPhone'),
+			enforcePhone: loadState('registration', 'enforcePhone'),
+			message: loadState('registration', 'message'),
+			password: loadState('registration', 'password'),
+			additionalHint: loadState('registration', 'additionalHint'),
+			requesttoken: getRequestToken(),
+			loginFormLink: loadState('registration', 'loginFormLink'),
+			isPasswordHidden: true,
+			passwordInputType: 'password',
+			submitting: false,
+		}
+	},
+
+	methods: {
+		togglePassword() {
+			if (this.passwordInputType === 'password') {
+				this.passwordInputType = 'text'
+			} else {
+				this.passwordInputType = 'password'
+			}
+		},
+		submit() {
+			// prevent sending the request twice
+			this.submitting = true
+			setTimeout(() => {
+				this.submitting = false
```

Comment:
We should either send a `showSuccess` here or change the text on the submit button, so screenreaders get notified about the fact that submitting worked.